packages/api/tests/entities/Friends.test.ts

```diff
@@ -0,0 +1,302 @@
+      expect(friend.requested).toBeDefined();
+      expect(friend.initiator).toEqual(input.fromId);
+      expect(friend.status).toEqual('pending');
+      expect(friend.since).toBeNull();
+    });
+
+    it('should create friend successfully, case 2: from < to', async () => {
+      let input = { fromId: me.id, toId: user2.id };
+
+      if (me.id > user2.id) {
+        input = { fromId: user2.id, toId: me.id };
+      }
+      const friend = await TestClient.createFriend(input);
+
+      expect(friend.id).toBeDefined();
+      expect(friend.user1Id).toEqual(input.toId);
+      expect(friend.user2Id).toEqual(input.fromId);
+      expect(friend.requested).toBeDefined();
+      expect(friend.initiator).toEqual(input.fromId);
+      expect(friend.status).toEqual('pending');
+      expect(friend.since).toBeNull();
+    });
+
+    it('should throw error if from and to already exist', async () => {
+      expect.assertions(1); // Expect there to be an error
+
+      const input = {
+        fromId: me.id,
+        toId: user2.id
+      };
+
+      await TestClient.createFriend(input);
+
+      try {
+        await TestClient.createFriend(input);
+      } catch (e) {
+        expect(e.message).toContain('UNIQUE constraint failed: friend.user1Id, friend.user2Id');
+      }
+    });
+
+    it('should throw error when add B and A if A and B are already exist', async () => {
+      expect.assertions(1); // Expect there to be an error
+
+      const input1 = {
+        fromId: me.id,
+        toId: user2.id
+      };
+
+      await TestClient.createFriend(input1);
+
+      const input2 = {
+        fromId: user2.id,
+        toId: me.id
+      };
+
+      try {
+        await TestClient.createFriend(input2);
+      } catch (e) {
+        expect(e.message).toContain('UNIQUE constraint failed: friend.user1Id, friend.user2Id');
+      }
+    });
+
+  });
+
+  describe('Query: getUserFriends', () => {
+    beforeEach(setup);
+    it('should return null since no friends', async () => {
+      const friends = await TestClient.getUserFriends(me.id);
+      expect(friends.length).toEqual(0);
+    });
+
+    it('should return one friend', async () => {
+      const input = {
+        fromId: me.id,
+        toId: user2.id
+      };
+      const addFriend = await TestClient.createFriend(input);
+      const friends = await TestClient.getUserFriends(me.id);
+
+      expect(friends.length).toEqual(1);
+      expect(friends[0].id).toEqual(addFriend.id);
+      expect(friends[0].user1Id).toEqual(addFriend.user1Id);
+      expect(friends[0].user2Id).toEqual(addFriend.user2Id);
+      expect(friends[0].since).toEqual(addFriend.since);
+      expect(friends[0].status).toEqual(addFriend.status);
+      expect(friends[0].requested).toEqual(addFriend.requested);
+      expect(friends[0].initiator).toEqual(addFriend.initiator);
+      expect(friends[0].user1.id).toEqual(addFriend.user1Id);
+      expect(friends[0].user2.id).toEqual(addFriend.user2Id);
```

Comment:
Can we use `toMatchObject` for all of these please :)? Much smaller code

Reply:
That goes for all the other places we do this

Reply:
getUserFriends test script has been changed to use toMatchObject.
But for others, e.g. the following one, I did not figure out how to use toMatchObject, especially the `requested` is a timestamp.
```
      expect(friend.id).toBeDefined();
      expect(friend.user1Id).toEqual(input.fromId);
      expect(friend.user2Id).toEqual(input.toId);
      expect(friend.requested).toBeDefined();
      expect(friend.initiator).toEqual(input.fromId);
      expect(friend.status).toEqual('pending');
      expect(friend.since).toBeNull();
```

Reply:
That's fine then